src/ir/function.rs

```diff
@@ -379,7 +385,7 @@ impl FunctionSig {
 
         // Don't parse operatorxx functions in C++
         let spelling = cursor.spelling();
-        if spelling.starts_with("operator") {
+        if OPERATOR_REGEX.is_match(&spelling) {
```

Comment:
I wonder if we should still check `starts_with` so that we don't do regex matching on every single function. In fact, I wonder if we could just avoid regexes all-together for this, by doing something like:

```rust
if spelling.starts_with("operator") && !clang::is_valid_identifier(&spelling) {
    return Err(ParseError::Continue);
}
```

(Or we could extract the "is valid in the middle of an identifier" check, and just check the one character after `operator`, to preserve behavior with the regex-based approach, though not sure that is worth it).

Reply:
I actually would prefer not to use regexes myself -- I was just preserving the previous implementation in #1292. I agree that we should do without. I will put the PR into draft until I rework it.

Reply:
This might be premature optimization--perhaps some benchmarks before? I think the regex crate optimizes patterns starting with a constant prefix:
https://github.com/rust-lang/regex/blob/65c4f8ee1f1c5ef21a287856f00b780eafbcd95c/PERFORMANCE.md#literals-in-your-regex-may-make-it-faster

Reply:
This does need measuring ideally... Though using regexes has the issue of duplicating the "is alphanumeric or an underscore" logic that `is_valid_identifier` already has, so perf is not (only) the reason for this.

Reply:
I would like to think we do not have to benchmark an implementation unless it is manifestly complex. I pushed a different implementation that does not use regexes.

Reply:
(I should say, sometimes we should definitely benchmark even a simple-looking implementation. I just do not think this is one of those times.)